controllers/resources/manager.go

```diff
@@ -116,7 +116,15 @@ func (m *manager) GenerateRemediationCRForMachine(machine *machinev1beta1.Machin
 		// So it can be ignored here.
 	}
 
-	return m.generateRemediationCR(machine.GetName(), mhcOwnerRef, machineOwnerRef, template)
+	cr, err := m.generateRemediationCR(machine.GetName(), mhcOwnerRef, machineOwnerRef, annotations.RemoveMultipleTemplatesAnnotation(template))
```

Comment:
it took me at least 20 minutes to find out why the annotation is lost in `generateRemediationCR`. Removing it by adding a function call in the method param here is... not ideal IMHO.
Actually I think removing the annotation is a bad thing in general. 
The logic in  `generateRemediationCR` should be changed to do the right thing.
Everything else will make maintenance very hard, because you have to look at multiple places to understand the outcome...